### DISCUSSION TITLE
Combobox when fail to retrieve dropdown content

### DISCUSSION THREAD

**Discussion by azhengtwilio** [twilio-labs/paste#1953] [General] [answered]
Hi, we have a few use cases where we need to make an API request to fill in the information for the dropdown of some comboboxes:
![Screen Shot 2021-10-14 at 5 02 31 PM](https://user-images.githubusercontent.com/83669618/137411461-403d1976-e32e-46ad-a0ca-b4a7cbd5d49d.png)

Is there a pattern for what we should show users when the API request fails and we don't have the data to display in the dropdown? 

Thank you!

**Comment by kate-porter** [comment, marked as answer]
Hey @azhengtwilio - great question. We don't have en established pattern for this quite yet (though we're always accepting contributions!). The most similar use case I could think of that exists in the product today is this empty state when searching for a page in the new Console nav:
![image](https://user-images.githubusercontent.com/5374045/137527987-04717fcb-99b6-4449-8221-a5b1269d7fbb.png)

What they essentially did was rendered that empty state message as a combobox dropdown item when the search query returned no results. That does mean that the empty state has a hover state and is clickable like a regular dropdown item. Combobox is designed this way because rendering the message as a dropdown item is the only way that users of assistive technology can access it (if the message were just plain text and not a combobox dropdown item, an AT user would have no way to navigate to it and would therefore not even know that it's there).

For your use case, you'll likely want to show an error message, so you could use styling similar to what we show for [inline errors on form elements](https://paste.twilio.design/components/input#input-with-inline-error). Do you have a product designer on your team who you could partner with to refine the design and copy of the error message?

**Reply by azhengtwilio** [reply]
yes, we do have a designer, I just talked to her about it. Thank you @katieporter
